src/helpers/helpers.config.ts

```diff
@@ -225,11 +272,16 @@ function _resolveScriptable(prop, value, target, receiver) {
   return value;
 }
 
-function _resolveArray(prop, value, target, isIndexable) {
+function _resolveArray(
+  prop: string,
+  value: unknown[],
+  target: ContextCache,
+  isIndexable: (key: string) => boolean
+) {
   const {_proxy, _context, _subProxy, _descriptors: descriptors} = target;
 
-  if (defined(_context.index) && isIndexable(prop)) {
-    value = value[_context.index % value.length];
+  if (typeof _context.index !== 'undefined' && isIndexable(prop)) {
```

Comment:
why not use the `defined` helper anymore?

Reply:
@kurkle `defined` returns `boolean`, it is not [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)